### PR TITLE
chore: added unit test coverage for assets

### PR DESCRIPTION
**Pull request by jopmiddelkamp**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

> [!NOTE]
> This pull request significantly expands and improves the unit test coverage for asset-related classes, particularly focusing on asset equality, validation, comparison, and serialization/deserialization logic. It also introduces a comprehensive new test class for `LiquidityPoolShareChangeTrustAsset`. These changes enhance the reliability of asset handling and ensure correct behavior in various scenarios, including edge cases and error handling.
> 
> **Expanded and improved test coverage for asset classes:**
> 
> * Added extensive tests to `AssetTest.cs` for asset creation, equality, validation (including error cases for invalid input), and comparison logic, ensuring robust coverage of various asset types and edge cases.
> * Added new tests to `AssetAmountTest.cs` to verify equality checks against different types and nulls, improving confidence in equality implementation.
> * Added comprehensive tests to `ChangeTrustAssetTest.cs` for asset creation, wrapper equality and comparison, serialization/deserialization (XDR), and error handling for unknown asset types.
> 
> **New test class for liquidity pool share assets:**
> 
> * Introduced `LiquidityPoolShareChangeTrustAssetTest.cs` with thorough tests covering construction, equality, comparison, serialization, and error handling for `LiquidityPoolShareChangeTrustAsset`.
> 
> **General improvements:**
> 
> * Updated test file imports to include necessary namespaces for new test scenarios and exception handling. [[1]](diffhunk://#diff-034ec5681ee5e52f25fc438f739ae8c26d77419e51b4438d4006b174b962847dL1-R6) [[2]](diffhunk://#diff-12ddf97bbdb0c26f8cfbab711e2eee2fb359c39d8a153f7dbd73de7cb016f300L1-R2) [[3]](diffhunk://#diff-76e5ebb3128548457ffc7c12bdc5e74de5f09a108faec03914d9d161b03fdc44L1-R5) [[4]](diffhunk://#diff-5aeb648a50c1cce1a0b16d5b122132646137bc0f03fcccbb8a806231defcf15aL1-R1)
> 
> These changes collectively strengthen the test suite for asset management, ensuring correct behavior and improved maintainability.

### CODE REVIEW

File: StellarDotnetSdk.Tests/Assets/AssetTest.cs [hunk 1]
```diff
@@ -105,4 +134,302 @@ public void TestAssetEquals()
...
+    /// </summary>
+    [TestMethod]
+    public void TestCompareTo_CreditAlphaNum4_SameCodeDifferentIssuer()
+    {
+        var issuer1 = KeyPair.Random().AccountId;
+        var issuer2 = KeyPair.Random().AccountId;
+        var credit1 = new AssetTypeCreditAlphaNum4("USD", issuer1);
+        var credit2 = new AssetTypeCreditAlphaNum4("USD", issuer2);
+        var comparison = credit1.CompareTo(credit2);
+        Assert.IsTrue(comparison != 0);
```
Comment (cuongph87) [inline StellarDotnetSdk.Tests/Assets/AssetTest.cs]: We can directly compare two assets, right?
```suggestion
        Assert.AreNotEqual(credit1, credit2);
```
